#!/usr/bin/env python
PACKAGE = "dynamic_tutorials"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 'name', 'type', 'level', 'description', 'default', 'min', 'max
gen.add("XY_GRID_RESOLUTION", float_t, 0, "Grid Resolution [m]", 1.0, 0, 100)
gen.add("YAW_GRID_RESOLUTION", float_t, 0, "Resolution of orientation in each cell [rad]", 30.0, 0, 100)
gen.add("MOTION_RESOLUTION", float_t, 0, "[m] path interporate resolution", 3.0, 0, 100)
gen.add("N_STEER", float_t, 0,  "number of steer command", 20.0, 0, 100)
gen.add("H_COST", float_t, 0, "A Heuristic cost", 5.0, 0, 100)
gen.add("VR", float_t, 0, "robot radius", 5.0, 0, 10)
gen.add("SB_COST", float_t, 0, "switch back penalty cost", 1.0, 0, 100)
gen.add("BACK_COST", float_t, 0, "backward penalty cost", 3.0, 0, 100)
gen.add("STEER_CHANGE_COST", float_t, 0, "steer angle change penalty cost", 5.0, 0, 100)
gen.add("STEER_COST", float_t, 0, "steer angle change penalty cost", 1.0, 0, 100)
gen.add("row_y_offset", float_t, 0, "steer angle change penalty cost", -3.0, 0, 10)
gen.add("threshold_distance_to_goal", float_t, 0, "steer angle change penalty cost", 1.0, 0, 10)
exit(gen.generate(PACKAGE, "base_trajectory_interface", "gpplanner_node_segment"))
               